### PR TITLE
feat(ShowMore): pass new state to the `onToggle` callback

### PR DESCRIPTION
**Pull request by TomasSlama**
[FX-3724]

### Description

We had [a discussion](https://toptal-core.slack.com/archives/CERF5NHT3/p1676299119242719) in Slack about users who would like to have information about the current state of the ShowMore component in `onToggle` callback

### How to test

- check updated unit tests
- play with the storybook

### Screenshots

n/a

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- n/a Make sure that additions and changes on design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests

**Breaking change**

- [x] codemod is created and showcased in the changeset
- [x] test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[FX-3723]: https://toptal-core.atlassian.net/browse/FX-3723?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[FX-3724]: https://toptal-core.atlassian.net/browse/FX-3724?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .changeset/nervous-chicken-trade.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+'@toptal/picasso': minor
+---
+
+---
+
+### ShowMore
+
+- the `onToggle` callback has a new state as a parameter
```
Comment (mkrl) [inline .changeset/nervous-chicken-trade.md:9]:
```suggestion
- add a new state as a parameter to `onToggle` callback
```

**Review by mkrl — APPROVED** [review]
:+1:

File: packages/picasso/src/ShowMore/ShowMore.tsx [hunk 2]
```diff
@@ -97,7 +97,7 @@ export const ShowMore = forwardRef<HTMLSpanElement, Props>(function ShowMore(
         <ButtonAction
           onClick={() => {
             setShownMore(!shownMore)
-            onToggle()
+            onToggle(!shownMore)
```
Comment (dmaklygin) [inline packages/picasso/src/ShowMore/ShowMore.tsx:100]: maybe small improvement: 
```
const nextState = !shownMore

setShownMore(nextState)
onToggle(nextState)
```

**Review by dmaklygin — APPROVED** [review]
👍

**Pull request merged**